src/test/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/integration/PlanControllerTest.kt

```diff
@@ -168,6 +169,22 @@ class PlanControllerTest : IntegrationTestBase() {
           .returnResult().responseBody
 
       assertThat(goalEntity?.targetDate).isNull()
+      assertThat(goalEntity?.goalStatus).isEqualTo(GoalStatus.FUTURE)
+    }
+
+    @Test
+    fun `should create goal with a target date`() {
+      val goalEntity: GoalEntity? =
+        webTestClient.post().uri("/plans/$mutablePlanUuid/goals").header("Content-Type", "application/json")
+          .headers(setAuthorisation(user = "Tom C", roles = listOf("ROLE_RISK_INTEGRATIONS_RO")))
+          .bodyValue(goalRequestBody)
+          .exchange()
+          .expectStatus().isCreated
+          .expectBody<GoalEntity>()
+          .returnResult().responseBody
+
+      assertThat(goalEntity?.targetDate).isNotNull()
+      assertThat(goalEntity?.goalStatus).isEqualTo(GoalStatus.ACTIVE)
     }
```

Comment:
Do we have a similar test for creating a goal without a targetDate? if so, maybe we want to update with a similar
`assertThat(goalEntity?.goalStatus).isEqualTo(GoalStatus.FUTURE)` assertion

Reply:
The test above this one checks for `GoalState.FUTURE`.